core/controllers/admin.py

```diff
@@ -107,11 +107,13 @@ class AdminPage(
     URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
     HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}
 
-    @acl_decorators.can_access_admin_page
-    def get(self) -> None:
+    # Here we use type Any because the sub-classes of 'Basehandler' can have
+    # 'get' method with different number of arguments and types.
+    @acl_decorators.open_access
+    def get(self, *args: Any, **kwargs: Any) -> None:
         """Renders the admin page."""
 
-        self.render_template('admin-page.mainpage.html')
+        self.render_template('oppia-root.mainpage.html')
```

Comment:
To establish a pattern: should we have a RootPageHandler in the backend (probably in base.py) and direct /admin and other routes to it, instead of creating a new boilerplate page handler for every page? Or will the "page" handlers for each page be different?

Reply:
I checked and we already have a oppia root handler so I reused that and deleted the AdminPageHandler entirely.